src/rdb/tests/rdbt.c

```diff
@@ -1145,6 +1166,82 @@ testm_disruptive_membership(crt_group_t *grp, uint32_t nranks,
 	return 0;
 }
 
+static int
+testm_dictate(crt_group_t *grp, uint32_t nranks, uint32_t nreplicas, uint64_t key, uint64_t val)
+{
+	int			rc;
+	d_rank_t		ldr_rank;
+	d_rank_list_t	       *ranks;
+	uint64_t		term;
+	uint64_t		val_out = 0;
+	struct rsvc_hint	h;
+	const int		NO_UPDATE = 0;
+	const int		UPDATE = 1;
+	int			i;
+
+	printf("\n==== TEST: RDB update, destroy majority, dictate, and lookup\n");
+
+	rc = rdbt_find_leader(grp, nranks, nreplicas, &ldr_rank, &term);
+	if (rc) {
+		fprintf(stderr, "ERR: RDB find leader failed\n");
+		return rc;
+	}
+	printf("INFO: RDB discovered leader rank %u, term="DF_U64"\n",
+	       ldr_rank, term);
+
+	rc = rdbt_test_rank(grp, ldr_rank, UPDATE, RDBT_MEMBER_NOOP, key, val,
+			    &val_out, &h);
+	if (rc) {
+		fprintf(stderr, "FAIL: update RDB failed via RPC to leader "
+				"rank %u: "DF_RC", hint:(r=%u, t="DF_U64"\n",
+				ldr_rank, DP_RC(rc), h.sh_rank, h.sh_term);
+		return rc;
+	}
+	if (val_out != val) {
+		fprintf(stderr, "FAIL: update val="DF_U64" expect "DF_U64"\n", val_out, val);
+		return -1;
+	}
+
+	ranks = d_rank_list_alloc(nreplicas);
+	if (ranks == NULL)
+		return -DER_NOMEM;
+
+	rc = dictate(grp, (ldr_rank + 1) % nranks, ldr_rank, ranks);
```

Comment:
Sorry for multiple questions in the same comment :)

1. So after this, will the pool end up with a single replica at the same leader rank as before?

2. Can the ldr_rank handle the RDBT_DICTATE RPC even when it is the "chosen" one to be the leader after dictate?

3. Can other test cases be considered such as (I guess assuming that the  rdb update has been replicated and committed), picking a chosen_rank other than ldr_rank, and then querying the new leader that was dictated in the dictate() call?

Reply:
> 1. So after this, will the pool end up with a single replica at the same leader rank as before?

Right.

> 2. Can the ldr_rank handle the RDBT_DICTATE RPC even when it is the "chosen" one to be the leader after dictate?

Yes. I think my last commit changing the "dist" operations have broken this test: The chosen rank is not stopped before the dictate operation begins. I'll fix the test.

> 3. Can other test cases be considered such as (I guess assuming that the  rdb update has been replicated and committed), picking a chosen_rank other than ldr_rank, and then querying the new leader that was dictated in the dictate() call?

Yes.

Reply:
> I think my last commit changing the "dist" operations have broken this test: The chosen rank is not stopped before the dictate operation begins. I'll fix the test.

Decided to change ds_rsvc_start instead, to stop a running service before letting it dictate.